File: tests/ClaudeToolbar.Core.Tests/Settings/NotificationSettingsTests.cs
using ClaudeToolbar.Core.Credentials;
using ClaudeToolbar.Core.Settings;

namespace ClaudeToolbar.Core.Tests.Settings;

public class NotificationSettingsTests
{
    [Fact]
    public void Defaults()
    {
        var n = AppSettings.CreateDefault().Notifications;
        Assert.True(n.Enabled);
        Assert.Equal(47831, n.Port);
        Assert.True(n.Sound);
    }

    [Fact]
    public void RoundTripAndMissingSection()
    {
        var s = AppSettings.CreateDefault();
        s.Notifications.Port = 50000;
        s.Notifications.Sound = false;
        var back = SettingsJson.Deserialize(SettingsJson.Serialize(s));
        Assert.Equal(50000, back.Notifications.Port);
        Assert.False(back.Notifications.Sound);
        Assert.Equal(47831, SettingsJson.Deserialize("{}").Notifications.Port);
    }

    [Fact]
    public void PortIsClamped()
    {
        var s = AppSettings.CreateDefault();
        s.Notifications.Port = 80;
        Assert.Equal(1024, SettingsValidator.Normalize(s).Notifications.Port);
        s.Notifications.Port = 70000;
        Assert.Equal(65535, SettingsValidator.Normalize(s).Notifications.Port);
    }

    [Fact]
    public void ClaudeSettingsPathFollowsTheConfigDir()
    {
        Assert.Equal("/Users/sam/.claude/settings.json", CredentialsPaths.ClaudeSettingsPath(null, "/Users/sam"));
        Assert.Equal("/tmp/cfg/settings.json", CredentialsPaths.ClaudeSettingsPath(" /tmp/cfg ", "/Users/sam"));
    }
}

File: mac/Tests/ClaudeToolbarCoreTests/Settings/NotificationSettingsTests.swift
import XCTest
@testable import ClaudeToolbarCore

final class NotificationSettingsTests: XCTestCase {
    func testDefaults() {
        let n = AppSettings.createDefault().notifications
        XCTAssertTrue(n.enabled)
        XCTAssertEqual(n.port, 47831)
        XCTAssertTrue(n.sound)
    }

    func testRoundTripAndMissingSection() throws {
        var s = AppSettings.createDefault()
        s.notifications.port = 50000
        s.notifications.sound = false
        let back = try SettingsJSON.decode(try SettingsJSON.encode(s))
        XCTAssertEqual(back.notifications.port, 50000)
        XCTAssertFalse(back.notifications.sound)
        XCTAssertEqual(try SettingsJSON.decode("{}").notifications.port, 47831)
    }

    func testPortIsClamped() {
        var s = AppSettings.createDefault()
        s.notifications.port = 80
        XCTAssertEqual(SettingsValidator.normalize(s).notifications.port, 1024)
        s.notifications.port = 70000
        XCTAssertEqual(SettingsValidator.normalize(s).notifications.port, 65535)
    }

    func testClaudeSettingsPathFollowsTheConfigDir() {
        XCTAssertEqual(CredentialsPaths.claudeSettingsPath(claudeConfigDir: nil, homeDirectory: "/Users/sam"), "/Users/sam/.claude/settings.json")
        XCTAssertEqual(CredentialsPaths.claudeSettingsPath(claudeConfigDir: " /tmp/cfg ", homeDirectory: "/Users/sam"), "/tmp/cfg/settings.json")
    }
}
